dag_in_context/src/from_egglog.rs

```diff
@@ -327,4 +327,33 @@ impl FromEgglog {
           _ => panic!("Invalid program: {:?}", program),
         })
     }
+
+    /// Converts a term back into a TreeProgram, but removes context nodes along the way.
+    /// This is crutial for the correctness of this conversion, since context nodes can break sharing
+    /// of the state edge.
+    /// TODO make extraction itself remove context nodes, solving this problem
+    pub fn program_from_egglog(&mut self, program: Term) -> TreeProgram {
+        let new_term = self.without_ctx_nodes(program);
+        self.program_from_egglog_preserve_ctx_nodes(new_term)
+    }
+
+    fn without_ctx_nodes(&mut self, expr: Term) -> Term {
+        match expr {
+            Term::App(head, children) => match (head.to_string().as_str(), children.as_slice()) {
+                ("InContext", [_assumption, expr]) => {
+                    self.without_ctx_nodes(self.termdag.get(*expr))
+                }
+                _ => {
+                    let mut new_children = vec![];
```

Comment:
nit: iter and collect might be easier to read

Reply:
had lifetime issues